backend/controllers/taskController.js

```diff
@@ -19,31 +19,32 @@ const createTask = asyncHandler(async (req, res) => {
 });
 
 const getTasks = asyncHandler(async (req, res) => {
-  const header = req.headers["authentication"];
+  // console.log("hereee");
+  const header = req.headers.authorization;
   const token = header.split(" ")[1];
 
   const user_data = jwt.decode(token);
 
-
   const userId = user_data._id;
-  const tasks = await Task.find({ user_id: userId }, (err, docs) => {
-    if(err) {
-      res.status(500);
-      throw new Error(`Could not fetch doc ${docs}`)
-    }
-  });
-
-  res.status(200).json(tasks);
+  await Task.find({ user_id: userId })
+    .then((docs) => {
+      res.status(200).json(docs);
+    })
+    .catch((err) => {
+      if (err) {
+        res.status(500);
+        throw new Error(`Could not fetch doc ${docs}`);
+      }
+    });
 });
 
 const getTasksById = asyncHandler(async (req, res) => {
   // Task id Parameter
-  const header = req.headers["authentication"];
+  const header = req.headers.authorization;
```

Comment:
Similar to my comment in the `authenticateToken.js` file, you no longer need to obtain the token from the header since the ID is already set in `req.id` from the middleware. Please make the changes to other files (controllers) that also do this as well and test the changes to ensure that everything is working as intented

Reply:
Fixed: Changes are reflected in recent commits